app/components/DatasetComponent.tsx

```diff
@@ -16,9 +15,24 @@ import CommitContainer from '../containers/CommitContainer'
 
 import { getComponentDisplayProps, StatusDot } from './ComponentList'
 
-import { ComponentStatus, SelectedComponent } from '../models/store'
+import { ComponentStatus, SelectedComponent, CommitDetails } from '../models/store'
+import Body from './Body'
+import { Details } from '../models/details'
+import { ApiActionThunk } from '../store/api'
+import { Action } from 'redux'
 
 interface DatasetComponentProps {
+  data: CommitDetails
+
+  // display details
+  details: Details
+
+  // seting actions
+  setDetailsBar: (details: {[key: string]: any}) => Action
```

Comment:
Use a [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt) instead of object with string keys.

```suggestion
  setDetailsBar: (details: Record<string,any>) => Action
```